client/components/data/document/index.jsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * External dependencies.
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import each from 'lodash/each';
+
+/**
+ * Internal dependencies.
+ */
+import {
+	setTitle,
+	setDescription,
+	addLink,
+	addMeta,
+	setUnreadCount
+} from 'state/document/actions';
+
+class Document extends Component {
+	componentWillMount() {
+		this.props.setTitle( this.props.title );
+		this.props.setDescription( this.props.description );
+		this.props.setUnreadCount( this.props.unreadCount );
+
+		if ( this.props.link ) {
+			each( this.props.link, ( link ) => {
+				this.props.addLink( link );
+			} );
+		}
+
+		if ( this.props.meta ) {
+			each( this.props.meta, ( meta ) => {
+				this.props.addMeta( meta );
+			} );
+		}
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.title !== nextProps.title ) {
+			this.props.setTitle( nextProps.title );
+		}
+
+		if ( this.props.description !== nextProps.description ) {
+			this.props.setDescription( nextProps.description );
+		}
+
+		if ( this.props.unreadCount !== nextProps.unreadCount ) {
+			this.props.setUnreadCount( nextProps.unreadCount );
+		}
+
+		if ( this.props.link !== nextProps.link ) {
+			each( nextProps.link, ( link ) => {
+				this.props.addLink( link );
+			} );
+		}
+
+		if ( this.props.meta !== nextProps.meta ) {
+			each( nextProps.meta, ( meta ) => {
+				this.props.addMeta( meta );
+			} );
+		}
+	}
+
+	render() {
+		return null;
+	}
+}
+
+Document.propTypes = {
+	title: PropTypes.string,
+	description: PropTypes.string,
+	unreadCount: PropTypes.number,
+	link: PropTypes.array,
+	meta: PropTypes.array,
+
+	setTitle: PropTypes.func.isRequired,
+	setDescription: PropTypes.func.isRequired,
+	addLink: PropTypes.func.isRequired,
+	addMeta: PropTypes.func.isRequired,
+	setUnreadCount: PropTypes.func.isRequired
+};
+
+Document.defaultProps = {
+	title: '',
```

Comment:
We can probably omit these default values. Or, at the very least, change the conditions above to not only check for the existence of array props, but also that length is set. Seems redundant that we check `if ( this.props.link ) {` if we're defaulting to an empty array anyways (which is truthy).